spacelift/resource_aws_integration.go

```diff
@@ -0,0 +1,176 @@
+package spacelift
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/shurcooL/graphql"
+
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/structs"
+)
+
+func resourceAWSIntegration() *schema.Resource {
+	return &schema.Resource{
+		Description: "" +
+			"`spacelift_aws_integration` represents an integration with an AWS " +
+			"account. This integration is account-level and needs to be explicitly " +
+			"attached to individual stacks in order to take effect." +
+			"\n\n" +
+			"Note: when assuming credentials for **shared worker**, Spacelift will use `$accountName-$integrationID@$stackID-suffix` " +
+			"or `$accountName-$integrationID@$moduleID-suffix` as [external ID](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html) " +
+			"and `$runID@$stackID@$accountName` truncated to 64 characters as [session ID](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole)," +
+			"Suffix will be read or write.",
+
+		CreateContext: resourceAWSIntegrationCreate,
+		ReadContext:   resourceAWSIntegrationRead,
+		UpdateContext: resourceAWSIntegrationUpdate,
+		DeleteContext: resourceAWSIntegrationDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			// Required.
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The friendly name of the integration",
+				Required:    true,
+			},
+			"role_arn": {
+				Type:        schema.TypeString,
+				Description: "ARN of the AWS IAM role to attach",
+				Required:    true,
+			},
+			"generate_credentials_in_worker": {
+				Type:        schema.TypeBool,
+				Description: "Generate AWS credentials in the private worker",
+				Optional:    true,
+				Default:     false,
+			},
+			// Optional.
+			"external_id": {
+				Type:        schema.TypeString,
+				Description: "Custom external ID (works only for private workers).",
+				Optional:    true,
+			},
+			"duration_seconds": {
+				Type:        schema.TypeInt,
+				Description: "Duration in seconds for which the assumed role credentials should be valid",
+				Default:     900,
+				Optional:    true,
+			},
+			"labels": {
+				Type:        schema.TypeSet,
+				Description: "Labels to set on the integration",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+			},
+			"legacy": {
```

Comment:
Thinking about this more, maybe we shouldn't include `legacy` on the resource or data source. I can't think of a scenario where the info would be useful within Terraform, and it's an extra thing to deprecate later on.